packages/grid/x-data-grid/src/DataGrid/DataGrid.tsx

```diff
@@ -568,19 +574,7 @@ DataGridRaw.propTypes = {
    * If some of the rows have children (for instance in the tree data), this number represents the amount of top level rows wanted on each page.
    * @default 100
    */
-  pageSize: chainPropTypes(PropTypes.number, (props: any) => {
-    if (props.pageSize && props.pageSize > MAX_PAGE_SIZE) {
-      return new Error(
```

Comment:
Same as https://github.com/mui/mui-x/pull/6773/files#r1025449627
`setPageSize` now throws an error if pageSize > 100 in `DataGrid`